File: python/6_kyu/multiples_of_3_or_5.py
# Multiples of 3 or 5
# https://www.codewars.com/kata/514b92a657cdc65150000006/train/python

def solution(num):
    ans = []

    for i in range(0, num):
        if i % 3 == 0 or i % 5 == 0:
            ans.append(i)

    return sum(ans)

# # Best Solution
# def solution(number):
#     return sum(x for x in range(number) if x % 3 == 0 or x % 5 == 0)

print(solution(10))

File: javascript/6_kyu/multiples_of_3_or_5.js
// Multiples of 3 or 5
// https://www.codewars.com/kata/514b92a657cdc65150000006/train/javascript

function solution(num) {
    var ans = [];

    for (var i = 0; i < num; i++) {
        if ( i % 3 == 0 || i % 5 == 0) {
            ans.push(i);
        }
    }

    return ans.reduce( function(a, b) { return a + b; }, 0);
}

console.log(solution(10));